conda_env/specs/yaml_file.py

```diff
@@ -0,0 +1,25 @@
+from .. import env
+from ..exceptions import EnvironmentFileNotFound
+
+
+class YamlFileSpec(object):
+    _environment = None
+
+    def __init__(self, filename=None, name=None, **kwargs):
+        self.filename = filename
+        self.name = name
```

Comment:
Is there a reason to store `name` here?  Since this takes `**kwargs`, we could just leave it in there and only pull out the relevant arguments.